framework/src/main/java/org/apache/drill/test/framework/TestDriver.java

```diff
@@ -328,6 +332,43 @@ public int runTests() throws Exception {
         }
       }
 
+      //TO check
+      if(cmdParam.threads >= tempTobeRunFailures.size()){
```

Comment:
make the comparison the other way round to make it more clear. Also add a comment. The idea here is that we run this random failure detection logic only when the number of failures are low. We are using the concurrency level as the maximum number of failures under which the logic kicks in.